File: pset6/readability/readability.py
import sys
while True:
    text = input("Text: ")

    words = 0
    letters = 0
    sentences = 0

    for i,v in enumerate(text):
        if str(v).isalpha() == True:
            letters += 1
        if v == ' ':
            words += 1
        if v == '.' or v == '!' or v == '?':
            sentences += 1

    words += 1

    L = (float(letters) / float(words)) * 100
    S = (float(sentences) / float(words)) * 100
    final = round(0.0588 * float(L) - 0.296 * float(S) - 15.8)
    #print(final)

    if final < 1:
        print("Before Grade 1")
    elif final >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {int(final)}")

    sys.exit(0)

File: pset2/readability/readability.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    //string input = get_string("Text: ");
    //string input = "There are more things in Heaven and Earth, Horatio, than are dreamt of in your philosophy.";
    string input = get_string("Text: ");

    int numberofletters, numberofwords, numberofsentences;
    numberofletters = 0;
    numberofwords = 0;
    numberofsentences =0;

    for(int i=0 ; i < strlen(input) ; i++)
    {
        if (isalpha(input[i]))

            numberofletters++;

        if (input[i] == ' ')

            numberofwords++;

        if ((input[i] == '.') || (input[i] == '!') || (input[i] == '?'))

            numberofsentences++;


    //printf("i is=%i  nol=%i\n",i ,numberofletters);
    //YARIN BURAYA DIKKATprintf("i is=%i  nol=%i\n",i ,numberofwords + 1);
    //printf("i is=%i  nol=%i\n",i ,numberofsentences);
    }
    numberofwords = numberofwords + 1;
    float L = (numberofletters / (float) numberofwords) * 100;
    float S = (numberofsentences / (float) numberofwords) * 100;
    int final = round(0.0588 * L - 0.296 * S - 15.8);
    //printf("%i", final);
    if (final < 1)
        printf("Before Grade 1\n");
    else if (final >= 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", final);



}

